packages/snap/src/index.ts

```diff
@@ -0,0 +1,463 @@
+import { OnRpcRequestHandler } from "@metamask/snaps-types";
+import { panel, text, heading, divider, copyable } from "@metamask/snaps-ui";
+import { initializeChains } from "./initialize";
+import { Chain, Chains, Fees } from "./types/chains";
+import { Address } from "./types/address";
+import { ChainState, AddressState } from "./state";
+import { Result } from "./types/result";
+import { submitTransaction } from "./transaction";
+import { getAddress } from "./address";
+
+/**
+ * Handle incoming JSON-RPC requests, sent through `wallet_invokeSnap`.
+ *
+ * @param args - The request handler args as object.
+ * @param args.request - A validated JSON-RPC request object.
+ * @returns A result object.
+ * @throws If the request method is not valid for this snap.
+ */
+export const onRpcRequest: OnRpcRequestHandler = async ({
+  request,
+}): Promise<Result> => {
+  let res: Object = {};
+  let confirmation: string | boolean | null = false;
+  switch (request.method) {
+    case "initialize":
+      // Ensure user confirms initializing Cosmos snap
+      confirmation = await snap.request({
+        method: "snap_dialog",
+        params: {
+          type: "confirmation",
+          content: panel([
+            text(
+              "Would you like to add Cosmos chain support within your Metamask wallet?"
+            ),
+          ]),
+        },
+      });
+      if (!confirmation) {
+        throw new Error("Initialize Cosmos chain support was denied.");
+      }
+      let chains = new Chains([]);
+      let chainList = await initializeChains();
+      chains = new Chains(chainList);
```

Comment:
Does `new Chains`  have some kind of side-effect?  If so, please document it, otherwise remove the dead code:
```suggestion
      let chainList = await initializeChains();
      let chains = new Chains(chainList);
```